src/radical/pilot/pytask.py

```diff
@@ -45,10 +45,15 @@ def get_func_attr(bson_obj):
         if not isinstance(bson_obj, str):
             raise ValueError('bson object should be string')
 
-        pytask = deserialize_bson(bson_obj)
-        func   = deserialize_obj(pytask['func'])
+        try:
+            pytask = deserialize_bson(bson_obj)
+            func   = deserialize_obj(pytask['func'])
+            args   = list(pytask['args'])
+            kwargs = pytask['kwargs']
+        except Exception as e:
+            raise ValueError('bson_obj is not a PythonTask') from e
```

Comment:
Aymen is looking into this already.  Since this is a performance critical code path I would like to avoid the additional try/except/raise, but you are also right, better error handling is needed also.  See later comment below.

Reply:
In principle, this should not raise any exceptions since the exception is expensive if it was raised [(python 3.9 docs),] and since the serializer will do that for us (i.e. raise an exception) then the `try/except` is not needed indeed. If it happens then the sterilizer (dill) will raise an exception that can fall naturally to the invoker.

Reply:
I suppose the question is whether you want to distinguish between errors from dill versus errors in the expectations on the object deserialized. It looked to me like I couldn't assume that KeyError would definitely _not_ come from `deserialize_bson` or `deserialize_obj`.